### PR TITLE
KOGITO-4185 DMN OAS definitions reference are broken due to Quarkus v1.11 q redirect

### PR DESCRIPTION
**Pull request by tarilabs**
Ref: https://issues.redhat.com/browse/KOGITO-4185

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: kogito-codegen/src/main/java/org/kie/kogito/codegen/decision/DecisionRestResourceGenerator.java [hunk 1]
```diff
@@ -200,7 +200,7 @@ private void processOASAnn(MethodDeclaration dmnMethod, DecisionService ds) {
             inputRef = withOASResult.getNamingPolicy().getRef(identifyInputSet);
             outputRef = withOASResult.getNamingPolicy().getRef(identifyOutputSet);
         }
-        final String DMN_DEFINITIONS_JSON = "dmnDefinitions.json";
+        final String DMN_DEFINITIONS_JSON = "/dmnDefinitions.json";
```
Comment (tarilabs) [inline kogito-codegen/src/main/java/org/kie/kogito/codegen/decision/DecisionRestResourceGenerator.java:203]: Ref: https://github.com/quarkusio/quarkus/issues/14307#issuecomment-760796091

File: integration-tests/integration-tests-quarkus-decisions/pom.xml [hunk 2]
```diff
@@ -81,6 +81,17 @@
...
     </dependency>
+    <dependency>
+      <groupId>io.swagger.parser.v3</groupId>
+      <artifactId>swagger-parser</artifactId>
+      <scope>test</scope>
+      <exclusions>
+        <exclusion> <!-- Kogito-on-Quarkus uses Kogito-Codegen's CP instead of PCL, so we need to manually exclude. -->
+          <groupId>io.swagger.core.v3</groupId>
+          <artifactId>swagger-annotations</artifactId>
+        </exclusion>
```
Comment (tarilabs) [inline integration-tests/integration-tests-quarkus-decisions/pom.xml:92]: Ref: https://issues.redhat.com/browse/KOGITO-4270

**Review by mariofusco — APPROVED** [review]

**Review by danielezonca — APPROVED** [review]
Thanks for the PR 👍

**Comment by danielezonca** [comment]
@jomarko 
FYI
Have a look to tests in this PR, we can have something similar with archetype tests too

**Review by hellowdan — APPROVED** [review]

**Pull request merged**